src/store/actions/authActions.ts

```diff
@@ -0,0 +1,43 @@
+import axios from 'axios';
+import { AsyncStorage } from 'react-native';
+import { secretIp } from '../../../secrets/secrets';
+import { fetchProfileAndTasksOnLogin } from './profileActions';
+import { AuthStateActions } from './loadStateActions';
+import { createCalendar } from './calendarActions';
+
+export const LOGIN = 'LOGIN';
+export const LOGOUT = 'LOGOUT';
+export const AUTH_ERROR = 'AUTH_ERROR';
+
+export const login = () => {
+    return {
+        type: LOGIN,
+    };
+};
+
+export const logout = () => {
+    return {
+        type: LOGOUT,
+    };
+};
+
+export const checkCredentials = ({ email, password }) => {
+    return async (dispatch) => {
+        await axios
+            .post(secretIp + `/api/authentication/login`, {
+                email: email,
+                password: password,
+            })
+            .then((response) => {
+                AsyncStorage.setItem('access_token', response.data.access_token);
+                dispatch(createCalendar());
+            })
+            .then(() => dispatch(fetchProfileAndTasksOnLogin()))
+            .then(() => dispatch(AuthStateActions.Loaded()))
+            .then(() => dispatch(login()))
```

Comment:
you're doing a lot here for something called `checkCredentials`. I don't think this logic belongs in `checkCredentials` although I may be wrong. Even if you always wanted to create calendar, fetch profile, fetch tasks, set loaded state, and login, i'm skeptical that it all belongs in this function. You are basically coupling all of these actions to checkCredentials which can potentially make it hard to re-use. It also seems to do more than just "checking credentials".